main.go

```diff
@@ -311,13 +311,9 @@ func main() {
 	}()
 
 	if enabled, _ := features.Enabled(features.GitManagedTransport); enabled {
-		managed.InitManagedTransport(ctrl.Log.WithName("managed-transport"))
-	} else {
-		if optimize, _ := feathelper.Enabled(features.OptimizedGitClones); optimize {
-			features.Disable(features.OptimizedGitClones)
-			setupLog.Info(
-				"disabling optimized git clones; git clones can only be optimized when using managed transport",
-			)
+		if err = managed.InitManagedTransport(ctrl.Log.WithName("managed-transport")); err != nil {
+			setupLog.Error(err, "could not register managed transports")
+			os.Exit(1)
```

Comment:
This introduces a hard stop when managed transport initialization fails. This is a new behavior compared to what we've done in the previous releases.
Highlighting so that we can discuss and decide if we really want to do this.

Reply:
Yes, but IMO it makes sense, since managed transports are the default now. So unless, users explicitly disable it, they would want to see the managed transports being used.

Reply:
This change removes the `enabled` package variable from pkg/git/libgit2/managed/init.go which was set at the end of `InitManagedTransport()` to indicate that managed transport was enabled successfully. But after removing that, we won't have any way to know if the managed transport is enabled or not, other than just trusting the feature flag values, which may not be accurate. This could be the reason to exit instead of continuing.

Reply:
I would say that only the reconciler should persistently error, instead of it disabling the processing of all source kinds.

Reply:
I don't mind this as well, but then I think that raises the argument that is this error much different from an error returned while starting a reconciler through `SetupWithManager()`? Asking because, we do error out if any of one the reconcilers fail to start. Now that managed transports are supposed to be the default and the intended way to checkout git repos, imo failing to register them, is equivalent to not being able to start the reconciler itself.

Reply:
I think that's another discussion to have. Given the recent insights on multi-tenancy (DoS) attack patterns, I think attempting to reconcile whatever we can (and possibly even making use of previous observed state backed by persistent storage) is more failure resistant than having the controller fail hard. It might be that we need to revise this in other areas as well, but within the context of this PR, I think this approach would be best.